pkg/dependenciesdistributor/dependencies_distributor.go

```diff
@@ -130,17 +130,41 @@ func (d *DependenciesDistributor) NeedLeaderElection() bool {
 
 func (d *DependenciesDistributor) discoverResources(period time.Duration) {
 	wait.Until(func() {
-		for _, gvr := range supportedTypes {
-			if d.InformerManager.IsHandlerExist(gvr, d.EventHandler) {
-				continue
+		newResources := lifted.GetDeletableResources(d.DiscoveryClientSet)
+		for gvr := range newResources {
+			if !d.InformerManager.IsHandlerExist(gvr, d.EventHandler) && !d.SkippedResourceConfig.GVRDisabled(gvr, d.RESTMapper) {
+				klog.Infof("Setup informer for %s", gvr.String())
+				d.InformerManager.ForResource(gvr, d.EventHandler)
 			}
-			klog.Infof("Setup informer for %s", gvr.String())
-			d.InformerManager.ForResource(gvr, d.EventHandler)
 		}
 		d.InformerManager.Start()
 	}, period, d.stopCh)
 }
 
+// Filter out cluster scope resources.
+func (d *DependenciesDistributor) Filter(obj interface{}) bool {
+	accessor, err := meta.Accessor(obj)
+	if err != nil {
+		return false
+	}
+	if accessor.GetNamespace() == metav1.NamespaceNone {
+		return false
+	}
+	typeAccessor, err := meta.TypeAccessor(obj)
+	if err != nil {
+		return false
+	}
+	referenceBinding, err := d.resourceBindingLister.ResourceBindings(accessor.GetNamespace()).Get(names.GenerateBindingName(typeAccessor.GetKind(), accessor.GetName()))
+	if err != nil {
+		return false
+	}
+	if len(referenceBinding.Spec.RequiredBy) > 0 {
+		return true
+	}
+	_, exist := referenceBinding.Annotations[bindingDependenciesAnnotationKey]
+	return exist
```

Comment:
Sorry, I do not get it. Why do you need to judge whether the key `bindingDependenciesAnnotationKey` exists? Could you explain your design to me?
BTW, I think `obj` should be a dependency resource, there never be `bindingDependenciesAnnotationKey` in its annotations.